Extension/src/LanguageServer/configurations.ts

```diff
@@ -684,7 +685,7 @@ export class CppProperties {
             this.parsePropertiesFile(); // Clear out any modifications we may have made internally.
             const config: Configuration | undefined = this.CurrentConfiguration;
             if (config) {
-                config.compileCommands = path;
+                config.compileCommands = [path];
```

Comment:
Question for my team. Do we want to prefer storing the compileCommands property as a string in the actual `c_cpp_properties.json` file, or as an array?  It doesn't matter how we set it here because the json will be reparsed in `handleConfigurationChange` and converted to an array.

One other thing to note is that we should change the Edit Configurations UI to make the `input` for compileCommands be a `textarea` instead, along with whatever associate changes should come with that. The answer to the question above would also guide the implementation for the storage of the data in the json file. (e.g. if we prefer string for the single entry case, we'd need to test for that when it comes via the UI and react accordingly)

Reply:
Quick suggestion for your first question, might be an overkill. Instead of forcing the value to be a string array both in `parsePropertiesFile()` and in `handleSquiggles()`, we create a "version 5" of the properties and add a function "updateToVersion5()" which actually saves a single entry as an array in the json file.   
Again, this might be overkill, just a suggestion.

Reply:
We have considered making changes to the version number, but this gets disruptive because many people commit their configurations to source control. We received a lot of negative feedback in the past when version changes were more frequent so we avoid it as best we can. 

We are technically breaking versioning and assuming some risk by not incrementing the version number with this change, but the addition is very subtle. We will need to discuss this as a team and decide if this is acceptable or not.

Reply:
@bobbrow Can't it be a string or an array in the c_cpp_properties.json?

Reply:
Should this thread be resolved now?